apps/aecore/src/aec_block_key_candidate.erl

```diff
@@ -66,44 +47,38 @@ adjust_target(Block, AdjHeaders) ->
     end.
 
 %% -- Internal functions -----------------------------------------------------
-int_create(BlockHash, Block, FeesInfo) ->
-    case aec_chain:get_block_state(BlockHash) of
-        {ok, Trees} ->
-            int_create(BlockHash, Block, FeesInfo, Trees);
-        error ->
-            {error, block_state_not_found}
-    end.
 
-int_create(BlockHash, Block, FeesInfo, Trees) ->
+int_create(BlockHash, Block) ->
     N = aec_governance:key_blocks_to_check_difficulty_count(),
     case aec_blocks:height(Block) < N of
         true  ->
-            int_create(BlockHash, Block, FeesInfo, Trees, []);
+            int_create(BlockHash, Block, []);
         false ->
             case aec_chain:get_n_generation_headers_backwards_from_hash(BlockHash, N) of
                 {ok, Headers} ->
-                    int_create(BlockHash, Block, FeesInfo, Trees, Headers);
+                    int_create(BlockHash, Block, Headers);
                 error ->
                     {error, headers_for_target_adjustment_not_found}
             end
     end.
 
-int_create(BlockHash, Block, FeesInfo, Trees, AdjChain) ->
+int_create(BlockHash, Block, AdjChain) ->
     case aec_keys:pubkey() of
         {ok, Miner} ->
-            int_create(BlockHash, Block, FeesInfo, Trees, Miner, AdjChain);
+            int_create(BlockHash, Block, Miner, AdjChain);
         {error, _} = Error ->
             Error
     end.
 
-int_create(PrevBlockHash, PrevBlock, FeesInfo, Trees, Miner, AdjChain) ->
-    {ok, Block, BlockInfo} = int_create_block(PrevBlockHash, PrevBlock, FeesInfo, Trees, Miner),
+int_create(PrevBlockHash, PrevBlock, Miner, AdjChain) ->
+    {ok, Trees} = aec_chain_state:calculate_keyblock_state(PrevBlockHash, Miner),
+    Block = int_create_block(PrevBlockHash, PrevBlock, Miner, Trees),
     case adjust_target(Block, AdjChain) of
-        {ok, AdjBlock} -> {ok, AdjBlock, BlockInfo#{ adj_chain => AdjChain }};
```

Comment:
Premature optimization - we will need that adjacent chain when implementing an optimized update for keyblocks...

Easy to re-introduce though.

Reply:
Please reintroduce it if needed. Currently it is not used.